storages/backends/s3.py

```diff
@@ -318,8 +318,17 @@ def __init__(self, **settings):
         self._bucket = None
         self._connections = threading.local()
 
-        if not self.config:
-            self.config = Config(
+        if self.config is not None:
+            warnings.warn(
+                "The 'config' class property is deprecated and will be "
+                "removed in a future version. Use AWS_S3_CLIENT_CONFIG "
+                "to customize any of the botocore.config.Config parameters.",
+                DeprecationWarning,
+            )
+            self.client_config = self.config
+
+        if self.client_config is None:
```

Comment:
I'm thinking that there is an unintended logical change here that makes `self.client_config` a required field, defaulted to `None`

Reply:
See the logic [here](https://github.com/jschneier/django-storages/blob/master/storages/base.py#L6-L11).

Reply:
I don't think I am following.  The `get_default_settings()` method [here](https://github.com/jschneier/django-storages/blob/8d7681fa22ae1532f03f144202340a8d124d3ccb/storages/base.py#L24) seems to return `{}` by default.

In our case, we have overridden this method as such:
```
    def get_default_settings(self):
        return {
            "access_key": setting(f'{self.setting_prefix}_ACCESS_KEY_ID'),
            "secret_key": setting(f'{self.setting_prefix}_SECRET_ACCESS_KEY'),
            ...,
        }
```

However, we don't have a setting for `client_config` because this is new.  This seems to be the root cause of the error: `object has no attribute 'client_config'`.  One solution I could try is to add a dummy attribute to the above method and default to `None`, but I shouldn't have to if the setting is not required, right?

Reply:
I don't provide backwards compat guarantees on `get_default_settings` as that is not a public API. If you want this to reliably work I'd recommend something like

```python
settings = super().get_default_settings()
...your changes here
```